src/nodes.coffee

```diff
@@ -1639,7 +1639,8 @@ exports.Op = class Op extends Base
     not @second
 
   isComplex: ->
-    not (@isUnary() and @operator in ['+', '-']) or @first.isComplex()
+    not (@isUnary() and @operator in ['+', '-'] and
+         @first instanceof Value and @first.isSimpleNumber())
```

Comment:
You're calling `Value::isSimpleNumber`. I say you don't do an `instanceof` test and instead use `Value::isComplex`. If `Value::isComplex` is returning `yes` for simple numbers, change its implementation.